pyracf/common/security_admin.py

```diff
@@ -170,13 +174,21 @@ def __raw_dump(self) -> None:
         if self.__debug:
             # Note, since the hex dump is logged to the console,
             # secrets will be redacted.
-            self._logger.log_hex_dump(raw_result_xml, self.__secret_traits)
+            self._logger.log_hex_dump(
+                raw_result_xml,
+                self.__secret_traits,
+                self._racf_key_map,
+                self._racf_message_key_map,
+            )
 
     # ============================================================================
     # Secrets Redaction
     # ============================================================================
     def __add_additional_secret_traits(self, additional_secret_traits: list) -> None:
         """Add additional fields to be redacted in logger output."""
+        unsupported_profile_types = ["permission", "groupConnection", "systemSettings"]
+        if self._profile_type in unsupported_profile_types:
+            return
```

Comment:
I wonder if we should be raising an exception if the additional secret the user provides is bad?

Reply:
That's a fair point, I will add an exception for this.

Reply:
Does this need it's own exception, or do you think that we could just using `ValueError` here? I just want to avoid us having dozens of exceptions for every possible case where the user provides a bad value. We may also want to revisit some of the trait errors as well. May make sense to use `ValueError` in those scenarios.